snuba/request/__init__.py

```diff
@@ -1,17 +1,12 @@
 from __future__ import annotations
 
-import itertools
-
 from collections import ChainMap
 from dataclasses import dataclass
 from deprecation import deprecated
 from typing import Any, Mapping
 
-from snuba.query.extensions import QueryExtension
 from snuba.query.query import Query
-from snuba.query.schema import GENERIC_QUERY_SCHEMA, SETTINGS_SCHEMA
 from snuba.request.request_settings import RequestSettings
-from snuba.schemas import Schema, validate_jsonschema
 
 
 @dataclass(frozen=True)
```

Comment:
I like the idea to put types in a dedicated module (that in general does not depend on anything else). I guess it makes more and more sense as long as we use more type checking because we are adding dependencies that were not explicit before. I don't have a strong opinion whether to use a types module or to use __init__. 
I think there is one tricky aspect though: it will not be possible to define in a clear cut way what goes in types and what does not. 
- type aliases are definitively there
- complex classes are definitively not there
- dataclass may be in between. Examples: https://github.com/getsentry/snuba/blob/master/snuba/datasets/schemas/join.py#L78-L95 .  This is a dataclass that has some logic inside as well. We have a few others. 

So, as long as we are ready to accept that some classes may be in a gray area, totally ok.